ls1intum/Artemis
# `Programming exercises`: Improve .gitignore files in project templates

### Checklist
#### General



### Motivation and Context


#6140 already tried to add the macOs file `.DS_Store` to the git ignore, but failed due to the request for more inclusions and inactivity from the original pull request creator. This PR now adds these requested changes.

### Description

Adds all files that are automatically created by IDE (Eclipse, IntelliJ, VSCode), editor (vim) or operating system (macOS, Windows and Linux) as suggested by GitHub itself (https://github.com/github/gitignore).
Languages that already used different approaches (e.g. VHDL and Assembler) by only including the specific files are untouched.
Swift already had an extensive gitignore for Xcode and is only improved by including the different operating systems.

### Steps for Testing


1. Look at the gitignore files and think about if any file should additionally be ignore or should not be ignored that currently are.
2. You can use the command `git check-ignore <fileNames>` in the terminal inside a git repository to check if the current gitignore is ignoring the listed files.

### Review Progress






#### Review
